src/components/animations/ButtonPressAnimation.js

```diff
@@ -68,280 +47,282 @@ const HapticFeedbackTypes = {
-    exclusive: true,
-    hapticType: HapticFeedbackTypes.selection,
-    minLongPressDuration: 500,
-    scaleTo: animations.keyframes.button.to.scale,
-  };
-
-  constructor(props) {
-    super(props);
-
-    this.clock = new Clock();
-    this.clockReversed = new Clock();
-    this.delayClock = new Clock();
-    this.gestureState = new Value(UNDETERMINED);
-    this.handle = undefined;
-    this.longPressDetected = false;
-    this.longPressTimeout = undefined;
-    this.scale = new Value(1);
-    this.shouldSpring = new Value(-1);
-
-    this.state = {
-      height: 0,
-      width: 0,
-    };
-
-    this.onGestureEvent = event([
-      {
-        nativeEvent: {
-          state: this.gestureState,
-        },
-      },
-    ]);
+  if (animatedRef.current == null) {
+    animatedRef.current = new Animated.Value(initialValue);
   }
 
-  componentWillUnmount = () => {
-    this.reset();
-  };
-
-  clearInteraction = () => {
-    if (this.props.isInteraction && this.handle) {
-      InteractionManager.clearInteractionHandle(this.handle);
-      this.handle = undefined;
-    }
-  };
-
-  clearLongPressListener = () => {
-    if (this.longPressTimeout) {
-      clearTimeout(this.longPressTimeout);
-    }
-  };
+  return animatedRef.current;
+}
 
-  createInteraction = () => {
-    if (this.props.isInteraction && !this.handle) {
-      this.handle = InteractionManager.createInteractionHandle();
-    }
-  };
+function useAnimatedClock() {
+  const animatedRef = React.useRef(null);
 
-  createLongPressListener = () => {
-    this.longPressDetected = false;
-    const { minLongPressDuration, onLongPress } = this.props;
-    if (onLongPress) {
-      this.longPressTimeout = setTimeout(
-        this.handleDetectedLongPress,
-        minLongPressDuration
-      );
-    }
-  };
+  if (animatedRef.current == null) {
+    animatedRef.current = new Animated.Clock();
+  }
 
-  handleHaptic = () => {
-    const { enableHapticFeedback, hapticType } = this.props;
+  return animatedRef.current;
+}
 
-    if (enableHapticFeedback) {
-      ReactNativeHapticFeedback.trigger(hapticType);
+export default function ButtonPressAnimation({
+  style,
+  children,
+  onPress,
+  scaleTo,
+  duration,
+  onPressStart,
+  easing,
+  transformOrigin,
+  isInteraction,
+  activeOpacity,
+  disabled,
+  onLongPress,
+  minLongPressDuration,
+  enableHapticFeedback,
+  hapticType,
+}) {
+  const prevGestureState = useAnimatedValue(0);
+  const gestureState = useAnimatedValue(0);
+  const scaleValue = useAnimatedValue(1);
+  const finished = useAnimatedValue(0);
+  const frameTime = useAnimatedValue(0);
+  const time = useAnimatedValue(0);
+  const toValue = useAnimatedValue(0.5);
+  const durationVal = useAnimatedValue(duration);
+  const animationState = useAnimatedValue(3);
+
+  const inteactionHandle = React.useRef();
+  const createHandle = React.useCallback(() => {
+    inteactionHandle.current = InteractionManager.createInteractionHandle();
+  });
```

Comment:
Can you include dependency list to all useCallbacks in this file (empty [] or with proper dependencies)